clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/DefaultIngestionBackend.java

```diff
@@ -164,13 +179,48 @@ public void setStorageEngineReference(
     }
   }
 
+  /**
+   * Create blob transfer manager and attach it to the ingestion backend since the creation of blob transfer manager is
+   * ad-hoc, depending on whether there are stores enabled with blob transfer.
+   * @param blobTransferManager
+   */
+  @Override
+  public void attachBlobTransferManager(BlobTransferManager blobTransferManager) {
```

Comment:
Do you think it would be simpler if we just always create a blob transfer manager to the ingestion backend even if we might not use it? (can still be turned off by a client config override) so that we don't need these logics here and there to do ad-hoc addition?

If you think this is actually more simple or legit, please don't hesitate to share

Reply:
I mean I do see you added a few more logics to try to create the manager based on if any of the workload is p2p enabled and such, so I am thinking if just creating it based on a config (default on) and use it when a ingestion task is current and p2p enabled.

Reply:
I think this is a good point actually.. Adding the manager in a Ad-hoc style does introduce some unnecessary complexity. Let me work on this.

Reply:
I will make it default 'off' for now since we are experimenting it